### ISSUE TITLE
HTreeMap.expireMaxSize(1000)

### ISSUE THREAD

**Issue by jankotek** [jankotek/MapDB#400]
Hi,

expireMaxSize(1000) is fairly small. HTreeMap size works as aproximation for concurrent reasons, so my guess is that this small size couses rounding errors and empties map on each iteration. 

I raised new issue to fix this in 2.0:

On Thursday, October 30, 2014 00:47:59 volker.stampa@eligotech.com wrote:

Hi,

to me it seems that a cache initialised with expireMaxSize and expireAfterWrite also expires 0ms after access. I tried the following (sorry for the scala-code, I hope it is clear enough):

```
      val map = db.createHashMap(UUID.randomUUID().toString).expireMaxSize(1000).expireAfterWrite(1, TimeUnit.DAYS).make[Int, Int]()
      map.put(1, 1)
//      println(map.get(1))
      Thread.sleep(1100) // wait for MapDB cleanup thread
      println(map.get(1))
```
This code prints 1, which is correct. However when the commented println is un-commented it prints:
```
1
null
```
So to me it seems that the first map.get records an access and the next time the cleanup-thread runs the 1 is evicted from the cache. When someone confirms that my assumptions are correct, I could open a bug.

Thanks
Volker

**Comment by jankotek** [comment]
Should be fixed by 016b7f8 . Related to #400. 

There was typo which swapped expiration queue head and tail on reopen.

**Comment by jankotek** [comment]
Fixed in both branches
